zerver/lib/events.py

```diff
@@ -844,9 +847,17 @@ def _draft_update_action(i: int) -> None:
                     if not was_admin and now_admin:
                         state["realm_bots"] = get_owned_bot_dicts(user_profile)
 
-            if client_gravatar and "avatar_url" in person:
+            if "avatar_url" in person:
                 # Respect the client_gravatar setting in the `users` data.
-                if person["avatar_url"].startswith("https://secure.gravatar.com"):
+                if (
+                    person["email_address_visibility"]
+                    != UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE
+                ):
```

Comment:
This change still needs to be discussed. We were thinking of not including the `email_address_visibility` value in the person objects of this event and for `realm_user/add` event above. One solution is to just check the `delivery_email` field, but in that case we would set `client_gravatar` as `True` for a user who has set `email_address_visibility` setting to `Admins only` and the acting user is admin and we would not sent avatar_url field in such case and expect clients to compute it from email.

This was not the case previously, as we did not send the`avatar_url` field only when email address visibility was set to `everyone`, i.e. the `avatar_url` field was sent to admin even if the visibility was set to `Admins only`. I am not sure whether to do this change or not as [this comment](https://github.com/zulip/zulip/blame/aa5536fe3046d76813015dd7db83297a213ca03a/zerver/tests/test_users.py#L361) mentions that we would need some changes for mobile app clients.

Reply:
Has this been discussing in an "api design" thread? If not, please start one and let's get the mobile team's opinion on it.

Reply:
Posted about it here: https://chat.zulip.org/#narrow/stream/378-api-design/topic/email.20address.20visibility/near/1497722.